tests/services/sessions/test_call_context.py

```diff
@@ -0,0 +1,107 @@
+import asyncio
+import time
+
+from jupyter_server.auth.utils import get_anonymous_username
+from jupyter_server.base.handlers import JupyterHandler
+from jupyter_server.services.kernels.kernelmanager import AsyncMappingKernelManager
+from jupyter_server.services.sessions.call_context import CallContext
+
+
+async def test_jupyter_handler_contextvar(jp_fetch, monkeypatch):
+    # Create some mock kernel Ids
+    kernel1 = "x-x-x-x-x"
+    kernel2 = "y-y-y-y-y"
+
+    # We'll use this dictionary to track the current user within each request.
+    context_tracker = {
+        kernel1: {"started": "no user yet", "ended": "still no user", "user": None},
+        kernel2: {"started": "no user yet", "ended": "still no user", "user": None},
+    }
+
+    # Monkeypatch the get_current_user method in Tornado's
+    # request handler to return a random user name for
+    # each request
+    async def get_current_user(self):
+        return get_anonymous_username()
+
+    monkeypatch.setattr(JupyterHandler, "get_current_user", get_current_user)
+
+    # Monkeypatch an async method in the mapping kernel manager.
+    # We chose a method that takes the kernel_id as as required
+    # first argument to ensure the kernel_id is correct in the
+    # current context.
+    async def shutdown_kernel(self, kernel_id, *args, **kwargs):
+        # Get the Jupyter Handler from the current context.
+        current: JupyterHandler = CallContext.get(CallContext.JUPYTER_HANDLER)
+        # Get the current user
+        context_tracker[kernel_id]["user"] = current.current_user
+        context_tracker[kernel_id]["started"] = current.current_user
+        time.sleep(1.0)
```

Comment:
We should update this to async:
```suggestion
        await asyncio.sleep(1.0)
```

Reply:
Oops. I forgot to commit this line.